### PR TITLE
Fix paths for Notebooks and Jupyter Lab

### PR DESCRIPTION
**Pull request by eap**
1) In the Jupyter notebooks [documentation](http://jupyter-notebook.readthedocs.io/en/stable/extending/contents.html) on ContentsManagers, it is noted that directory paths have their leading and trailing slashes stripped. This library violated that expecation and caused a major user-facing bug in Jupyter Notebooks where directories would have to be clicked multiple times to create a correctly formatted request. This change resolves the discrepancy by stripping trailing forward slashes from directory paths.

2) Jupyter Lab does not pass the type of a model to all `get` calls. This ambiguous case was previously handled by assuming untyped requests are either a blob or a notebook. When they are directories, 404 is returned. This change adds more advanced type resolution for cases of ambiguous type.

3) I've begin a small bit of code cleanup by creating an init function where expected member variables can be pre-set.

### CODE REVIEW

**Comment by eap** [comment]
Notes on testing: 
I've run unit tests in 2.7 and 3.6. The change is fairly small, I don't have any reason to believe that it would break other versions.

I've manually tested in Python 3.6 and 2.7 with both Jupyter Lab and Jupyter Notebooks using the following versions:
 * jupyter-core==4.4.0
 * jupyter-client=5.2.2
 * jupyterlab==0.31.11

File: jgscm/__init__.py [hunk 1]
```diff
@@ -428,29 +441,50 @@ def client(self):
...
+                                    model=model,
+                                    contents_manager=self)
             except Exception:
                 self.log.error("Post-save hook failed on %s", os_path, exc_info=True)
 
     @default("checkpoints_class")
     def _checkpoints_class_default(self):
         return GoogleStorageCheckpoints
 
+    def _resolve_storagetype(self, path, type):
```
Comment (vmarkovtsev) [inline jgscm/__init__.py]: Could you please rename `type` to smth else - it clashes with the built-in name. I know that it does not matter much in this particular case, but still.

File: jgscm/__init__.py [hunk 2]
```diff
@@ -209,6 +209,10 @@ class GoogleStorageContentManager(ContentsManager):
             """
                          )
 
+    def __init__(self, *args, **kwargs):
+        self._client = None
```
Comment (vmarkovtsev) [inline jgscm/__init__.py]: Would be nice to have a comment about what is this client

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]
Thanks for the contribution! LGTM overall, just a minor quirk with the arg name.

File: jgscm/__init__.py (hunk 1 above)
Comment (eap) [reply jgscm/__init__.py]: Was named type to comply with existing arg-name convention. I agree that overriding a builtin is... unpleasant. Anyways. Fixed.

File: jgscm/__init__.py (hunk 2 above)
Comment (eap) [reply jgscm/__init__.py]: Done

**Review by vmarkovtsev — APPROVED** [review]

**Pull request merged**